### DISCUSSION TITLE
How to use native functionalities with WinUI?

### DISCUSSION THREAD

**Discussion by BraveEvidence** [microsoft/WinUI-Gallery#962] [Q&A] [answered]
I am a newbie in windows app development. I have heard great things about winui and I am learning it. I am also keeping a close eye on windows app sdk. Suppose there is some native functionality like audio, bluetooth, video, sqlite etc which I need to integrate in my winui app and if this functionality is not available in windows app sdk, does the integration of native functionality remains the same to that of WPF, UWP etc. If I have some tutorial on integrating bluetooth with WPF, UWP etc, can I use the same tutorial with winui as well only the Ui logic will change correct?

**Comment by marcelwgn** [comment, marked as answer]
Generally speaking yes. Unlike UWP, WinUI 3 apps also can use the Win32 APIs that are usable in Winforms, WPF, etc. So if you have a tutorial for WPF that shows how to access Bluetooth APIs you should be able to do that since WinUI 3 only is the UI API.
